### ISSUE TITLE
How should I put weight on custom loss function for Cost sensitive learning.

### ISSUE THREAD

**Issue by minsung-k** [keras-team/keras#15775]
Hi, I have a question.
How should I put the weight that only I wanted?

In classification, there are 4 groups, TP, TN, FP, and FN.

if I made my custom_loss like this.
custom_loss = (1-y_true) * y_pred * (cost_1) + y_true * y_pred * (cost_2)

If I made loss like it, it is impossible only to put weight in FP, because they don't divide yet.

In this case, how should I put weight for cost-sensitive learning?

**Comment by sanatmpa1** [comment]
@minsung-k,

Can you take a look at this [SO thread](https://stackoverflow.com/questions/52695913/custom-loss-function-in-keras-to-penalize-false-negatives)? though the thread is about penalizing FN more, but it can be a reference for you to start with. Thanks!

**Comment by minsung-k** [comment]
@sanatmpa1 

Thank you for your reply.

Btw, is it works?

I think it doesn't work.
This is because y_pred should look like a continuous number between 0~1. not 0 or 1.

So, 

    TN = np.logical_and(K.eval(y_true) == 0, K.eval(y_pred) == 0)
    TP = np.logical_and(K.eval(y_true) == 1, K.eval(y_pred) == 1)

    FP = np.logical_and(K.eval(y_true) == 0, K.eval(y_pred) == 1)
    FN = np.logical_and(K.eval(y_true) == 1, K.eval(y_pred) == 0)

doesn't work. 

Could you help me how to handle this problem?

**Comment by minsung-k** [comment]
@google-ml-butler



> @minsung-k,
> 
> Can you take a look at this [SO thread](https://stackoverflow.com/questions/52695913/custom-loss-function-in-keras-to-penalize-false-negatives)? though the thread is about penalizing FN more, but it can be a reference for you to start with. Thanks!


Thank you!
It helps a lot

**Comment by sanatmpa1** [comment]
Thank you for your update. I am moving this issue to closed status, as it is resolved.
